lib/impure/db_sqlite.nim

```diff
@@ -716,8 +719,60 @@ proc setEncoding*(connection: DbConn, encoding: string): bool {.
 proc finalize*(sqlPrepared:SqlPrepared){.discardable.} = 
   discard finalize(sqlPrepared.PStmt)
 
+proc bindParam*(ps: SqlPrepared, paramIdx: int, val: int32):int =
```

Comment:
The `db_*` family of modules tries to abstract away vendor-specific error codes. So instead of returning the error code, you should throw an exception here with the error.

Reply:
I add a `dbBindParamError` template,
but I dunno how to display val, val could be seq[char] and very long.
`e.msg = "error binding param in position " & $paramIdx `

Reply:
You don't have to, just say the position and the vendor-specific error message.

Reply:
In that proc, does not need db context, it's hard to decide pass db context only for get db error message.

Reply:
`sqlite3_errstr` should work, though I can see that the interface is rather new :/

Maybe we can skip on the vendor-specific error message for now.

Reply:
ah , that's a nice api, considering in next PR.

Reply:
We can actually go ahead and store `DbConn` inside `SqlPrepared` (which will become an object, preferably with the fields exported).

It would just be easier if sqlite provides access to the db handle associated with the prepared statement...

Reply:
may handle error in lib level, as `db_postgres` already declared `SqlPrepared* = distinct string` ,and easier unify the dbs api.